src/Confluent.Kafka/Producer.cs

```diff
@@ -807,6 +834,22 @@ internal Producer(ProducerBuilder<TKey, TValue> builder)
             }
         }
 
+        private TopicPartition GetPartition(TopicPartition topicPartition, Message<TKey, TValue> message, TKey key, byte[] keyBytes, TValue value, byte[] valBytes)
```

Comment:
librdkafka will call the partitioner as needed, you should not forcibly call it from the dotnet code.